tests/test_index.c

```diff
@@ -12,564 +16,1028 @@
+			1), LZMA_PROG_ERROR);
+
+	lzma_index *index = lzma_index_init(NULL);
+	assert_true(index != NULL);
+
+	// Test with invalid unpadded size
+	assert_lzma_ret(lzma_index_append(index, NULL,
+			UNPADDED_SIZE_MIN - 1, 1), LZMA_PROG_ERROR);
+	assert_lzma_ret(lzma_index_append(index, NULL,
+			UNPADDED_SIZE_MAX + 1, 1), LZMA_PROG_ERROR);
+
+	// Test with invalid uncompressed size
+	assert_lzma_ret(lzma_index_append(index, NULL,
+			UNPADDED_SIZE_MAX, LZMA_VLI_MAX + 1),
+			LZMA_PROG_ERROR);
+
+	// Test expected successful block appends
+	assert_lzma_ret(lzma_index_append(index, NULL, UNPADDED_SIZE_MIN,
+			1), LZMA_OK);
+	assert_lzma_ret(lzma_index_append(index, NULL,
+			UNPADDED_SIZE_MIN * 2,
+			2), LZMA_OK);
+	assert_lzma_ret(lzma_index_append(index, NULL,
+			UNPADDED_SIZE_MIN * 3,
+			3), LZMA_OK);
+
+	lzma_index_end(index, NULL);
+
+	// Test uncompressed size growing too large.
+	// Should result in LZMA_DATA_ERROR
+	index = lzma_index_init(NULL);
+
+	assert_lzma_ret(lzma_index_append(index, NULL, UNPADDED_SIZE_MAX,
+			1), LZMA_DATA_ERROR);
+
+	// Test compressed size growing too large.
+	// Should result in LZMA_DATA_ERROR
+	assert_lzma_ret(lzma_index_append(index, NULL,
+			UNPADDED_SIZE_MIN, LZMA_VLI_MAX), LZMA_OK);
+	assert_lzma_ret(lzma_index_append(index, NULL,
+			UNPADDED_SIZE_MIN, 1), LZMA_DATA_ERROR);
+
+	// Currently not testing for error case when the size of the index
+	// grows too large to be stored. This was not practical to test for
+	// since too many blocks needed to be created to cause this.
+
+	lzma_index_end(index, NULL);
 }
 
 
-static lzma_index *
-create_big(void)
+static void
+test_lzma_index_stream_flags(void)
 {
-	lzma_index *i = lzma_index_init(NULL);
-	expect(i != NULL);
-
-	lzma_vli total_size = 0;
-	lzma_vli uncompressed_size = 0;
-
-	// Add pseudo-random sizes (but always the same size values).
-	uint32_t n = 11;
-	for (size_t j = 0; j < BIG_COUNT; ++j) {
-		n = 7019 * n + 7607;
-		const uint32_t t = n * 3011;
-		expect(lzma_index_append(i, NULL, t, n) == LZMA_OK);
-		total_size += (t + 3) & ~LZMA_VLI_C(3);
-		uncompressed_size += n;
-	}
+	// Only trivial tests done here testing for basic functionality.
+	// More in-depth testing for this function will be done in
+	// test_lzma_index_checks
+
+	// Testing for NULL inputs
+	assert_lzma_ret(lzma_index_stream_flags(NULL, NULL),
+			LZMA_PROG_ERROR);
+
+	lzma_index *index = lzma_index_init(NULL);
+	assert_true(index != NULL);
 
-	expect(lzma_index_block_count(i) == BIG_COUNT);
-	expect(lzma_index_total_size(i) == total_size);
-	expect(lzma_index_uncompressed_size(i) == uncompressed_size);
-	expect(lzma_index_total_size(i) + lzma_index_size(i)
-				+ 2 * LZMA_STREAM_HEADER_SIZE
-			== lzma_index_stream_size(i));
+	assert_lzma_ret(lzma_index_stream_flags(index, NULL),
+			LZMA_PROG_ERROR);
 
-	return i;
+	lzma_stream_flags stream_flags = {
+		.backward_size = LZMA_BACKWARD_SIZE_MIN,
+		.check = LZMA_CHECK_CRC32
+	};
```

Comment:
While the only valid value for .version is currently 0 and implicit initalization sets it to 0, it's good to explicitly set .version = 0.

The functions in index.c don't care about .backward_size but it is fine to set it here still.